### PR TITLE
Share Model URL fix

### PR DESCRIPTION
**Pull request by sayamkanwar**
Hi,
I have fixed the bug in the share model feature which generated different URLs every time. Now it generates only one URL for a model.

Here's a demo

share_patch

Please review @Ram81 @RishabhJain2018 
Thank you! :)

### CODE REVIEW

**Comment by sayamkanwar** [comment]
Also @Ram81, what else does the task demand? Wasn’t that the only correction in this task?

File: ide/views.py [hunk 1]
```diff
@@ -89,32 +89,51 @@ def save_to_db(request):
         net_name = request.POST.get('net_name')
         user_id = request.POST.get('user_id')
         next_layer_id = request.POST.get('nextLayerId')
-        public_sharing = True
         user = None
+        public_sharing = True
```
Comment (Ram81) [inline ide/views.py]: Why was this necessary?

File: ide/views.py [hunk 2]
```diff
@@ -89,32 +89,51 @@ def save_to_db(request):
...
-            model_update = NetworkUpdates(network_version=model_version,
-                                          updated_data=json.dumps({'nextLayerId': next_layer_id}),
-                                          tag='ModelShared')
-            model_update.save()
-
-            return JsonResponse({'result': 'success', 'id': model.id})
-        except:
-            return JsonResponse({'result': 'error', 'error': str(sys.exc_info()[1])})
+
+        if Network.objects.filter(name=net_name).exists():
```
Comment (Ram81) [inline ide/views.py]: What if filter returns `null`

File: ide/views.py [hunk 3]
```diff
@@ -89,32 +89,51 @@ def save_to_db(request):
...
+            # Update the exising json field
+            try:
+                if user_id:
+                    user_id = int(user_id)
+                    user = User.objects.get(id=user_id)
+                # load the model with the net name
+                model = Network.objects.get(name=net_name)
```
Comment (Ram81) [inline ide/views.py]: Why are we using `net_name` as a unique key?

File: ide/views.py [hunk 4]
```diff
@@ -89,32 +89,51 @@ def save_to_db(request):
...
+                model_id = model.id
```
Comment (Ram81) [inline ide/views.py]: Isn't there any better way to handle this for anonymous sessions?

**Comment by sayamkanwar** [comment]
@Ram81, I'm sorry I had to abandon this task as I could not understand what to do in this. But I would like to work on this after the contest too.

**Comment by Ram81** [comment]
Closing this as student has abandoned the task

File: ide/views.py (hunk 3 above)
Comment (sayamkanwar) [reply ide/views.py]: We are using ``net_name`` as a unique key because the only similarity between the new model and already existing model would be the model name. That's why I can't use anything else like id.

File: ide/views.py (hunk 4 above)
Comment (sayamkanwar) [reply ide/views.py]: Answered above

File: ide/views.py (hunk 2 above)
Comment (sayamkanwar) [reply ide/views.py]: That's why there is an if statement, if this returns true then this will be executed. Else it will create a new model record.

Comment (sayamkanwar) [reply ide/views.py]: If the result is null, it will return false

File: ide/views.py (hunk 1 above)
Comment (sayamkanwar) [reply ide/views.py]: This was necessary because then the model would not be shared and hence the canvas will not render the comments and sharing stuff on the model.
